stock_picking_batch_oca/models/stock_batch_picking.py

```diff
@@ -0,0 +1,243 @@
+        states={'draft': [('readonly', False)]},
+        default=lambda self: self.env['ir.sequence'].next_by_code(
+            'stock.picking.batch'
+        ),
+    )
+
+    # added state to be compatible with picking_ids
+    state = fields.Selection(
+        selection_add=[
+            ('assigned', 'Available'),
+        ],
+        string='State',
+        readonly=True, index=True,
+        help='the state of the batch picking. '
+             'Workflow is draft -> in_progress -> done or cancel',
+    )
+
+    date = fields.Date(
+        string='Date',
+        required=True, readonly=True, index=True,
+        states={
+            'draft': [('readonly', False)],
+            'in_progress': [('readonly', False)]
+        },
+        default=fields.Date.context_today,
+        help='date on which the batch picking is to be processed',
+    )
+
+    user_id = fields.Many2one(
+        comodel_name='res.users',
+        string='Picker',
+        readonly=True, index=True,
+        states={
+            'draft': [('readonly', False)],
+            'in_progress': [('readonly', False)]
+        },
+        help='the user to which the pickings are assigned',
+        old_name='picker_id',
+    )
+
+    use_oca_batch_validation = fields.Boolean(
+        default=lambda self: self.env.user.company_id.use_oca_batch_validation,
+        copy=False,
+    )
+
+    picking_ids = fields.One2many(
+        string='Pickings',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        help='List of picking managed by this batch.',
+    )
+    # TODO add comment to this field
+    active_picking_ids = fields.One2many(
+        string="Active Pickings",
+        comodel_name='stock.picking',
+        inverse_name='batch_id',
+        readonly=True,
+        domain=[('state', 'not in', ('cancel', 'done'))],
+    )
+
+    notes = fields.Text('Notes', help='free form remarks')
+
+    move_lines = fields.Many2many(
+        comodel_name='stock.move',
+        readonly=True,
+        string='Operations',
+        compute='_compute_move_lines',
+    )
+
+    move_line_ids = fields.Many2many(
+        comodel_name='stock.move.line',
+        string='Detailed operations',
+        compute='_compute_move_line_ids',
+        # HACK: Allow to write sml fields from this model
+        inverse=lambda self: self,
+    )
+
+    entire_package_ids = fields.Many2many(
+        comodel_name='stock.quant.package',
+        compute='_compute_entire_package_ids',
+        help='Those are the entire packages of a picking shown in the view of '
+             'operations',
+    )
+
+    entire_package_detail_ids = fields.Many2many(
+        comodel_name='stock.quant.package',
+        compute='_compute_entire_package_ids',
+        help='Those are the entire packages of a picking shown in the view of '
+             'detailed operations',
+    )
+
+    @api.depends('picking_ids')
+    def _compute_move_lines(self):
+        for batch in self:
```

Comment:
Should we prevent to compute fields if `use_oca_batch_validation` is not set?

Reply:
no harm from it, but updated